File: android/app/src/main/java/com/ankianywhere/services/SpacedRepetitionService.kt
package com.ankianywhere.services

class SpacedRepetitionService {
    data class ReviewResult(
        val easeFactor: Double,
        val interval: Int,
        val repetitions: Int
    )
    
    // SM-2 算法实现
    fun calculateNextReview(
        currentEaseFactor: Double,
        currentInterval: Int,
        currentRepetitions: Int,
        quality: Int
    ): ReviewResult {
        var easeFactor = currentEaseFactor
        var interval = currentInterval
        var repetitions = currentRepetitions
        
        if (quality >= 3) {
            // 回答正确
            interval = when (repetitions) {
                0 -> 1
                1 -> 6
                else -> (interval * easeFactor).toInt()
            }
            repetitions += 1
        } else {
            // 回答错误，重置
            repetitions = 0
            interval = 1
        }
        
        // 更新ease factor
        easeFactor += (0.1 - (5 - quality) * (0.08 + (5 - quality) * 0.02))
        if (easeFactor < 1.3) {
            easeFactor = 1.3
        }
        
        return ReviewResult(easeFactor, interval, repetitions)
    }
    
    companion object {
        val instance = SpacedRepetitionService()
    }
}

File: ios/AnkiAnywhere/AnkiAnywhere/Services/SpacedRepetitionService.swift
import Foundation

class SpacedRepetitionService {
    static let shared = SpacedRepetitionService()
    
    private init() {}
    
    // SM-2 算法实现
    func calculateNextReview(
        currentEaseFactor: Double,
        currentInterval: Int,
        currentRepetitions: Int,
        quality: Int
    ) -> (easeFactor: Double, interval: Int, repetitions: Int) {
        var easeFactor = currentEaseFactor
        var interval = currentInterval
        var repetitions = currentRepetitions
        
        if quality >= 3 {
            // 回答正确
            if repetitions == 0 {
                interval = 1
            } else if repetitions == 1 {
                interval = 6
            } else {
                interval = Int(Double(interval) * easeFactor)
            }
            repetitions += 1
        } else {
            // 回答错误，重置
            repetitions = 0
            interval = 1
        }
        
        // 更新ease factor
        easeFactor = easeFactor + (0.1 - Double(5 - quality) * (0.08 + Double(5 - quality) * 0.02))
        if easeFactor < 1.3 {
            easeFactor = 1.3
        }
        
        return (easeFactor, interval, repetitions)
    }
    
    func getNextReviewDate(interval: Int) -> Date {
        return Calendar.current.date(byAdding: .day, value: interval, to: Date()) ?? Date()
    }
}
